[One Workflow] Implement cancel-in-progress collision strategy

h88 opened the pull request:
## Overview

This PR implements the `cancel-in-progress` collision strategy for workflow concurrency control. When a concurrency limit is reached, this strategy cancels the oldest running execution(s) to make room for the new execution, allowing it to proceed immediately.

## Changes

### Core Implementation
- **ConcurrencyManager**: Implements the `cancel-in-progress` strategy logic
  - Queries for non-terminal executions in the same concurrency group
  - Calculates how many executions need to be cancelled
  - Cancels the oldest executions by updating their status to `CANCELLED`
  - Propagates cancellation via `forceRunIdleTasks()` to abort running tasks

### Repository Method
- **WorkflowExecutionRepository.getRunningExecutionsByConcurrencyGroup()**: New query method
  - Filters by `concurrencyGroupKey` 
  - Excluding terminal statuses (COMPLETED, FAILED, CANCELLED, SKIPPED, TIMED_OUT)

## Expected Results (Acceptance Criteria)

✅ When limit is reached, oldest non-terminal execution(s) are cancelled  
✅ Only the correct number of executions are cancelled (e.g., if max=2, active=3, cancel 2 oldest)  
✅ Cancelled executions have status=CANCELLED, cancelRequested=true, cancellationReason set  
✅ New execution proceeds immediately after cancelling old ones  
✅ Cancelled executions detect cancellation and abort gracefully  
✅ Works with concurrency.max values > 1  
✅ Query correctly finds non-terminal executions by concurrency group key  

## Testing

- Unit tests for `checkConcurrency()` with cancel-in-progress strategy
- Unit tests for query logic (`getRunningExecutionsByConcurrencyGroup`)
- Integration tests: cancel single execution when limit reached
- Integration tests: cancel multiple executions when needed
- Integration tests: verify cancelled executions abort correctly
- Integration tests: verify new execution proceeds after cancellation

## Leave it Better

### Dependency Injection & Architecture Improvements
- **ConcurrencyManager Dependency Injection**: Refactored `ConcurrencyManager` to use proper dependency injection pattern
  - Injected `WorkflowTaskManager` and `WorkflowExecutionRepository` as constructor dependencies
  - Removed these dependencies from method parameters, making the API cleaner and more testable
  - Created singleton instances in `start()` method for efficient reuse
  - `WorkflowTaskManager` and `WorkflowExecutionRepository` are created once in `start()` and reused throughout
  - Eliminates redundant instantiation and ensures consistent state management

###  Performance Optimizations
- **Bulk Update Implementation**:
  - Added bulkUpdateWorkflowExecutions() method to WorkflowExecutionRepository
  - Uses Elasticsearch bulk API for efficient batch updates
  - Single refresh: true operation for entire bulk instead of N individual refreshes
  - Follows existing pattern (Array<Partial<EsWorkflowExecution>>) consistent with bulkUpsert and updateWorkflowExecution
  
- **ID-Only Query Optimization**:
  - Refactored getRunningExecutionsByConcurrencyGroup() to return string[] instead of full EsWorkflowExecution[]
  - Uses _source: ['id'] to only fetch ID field from Elasticsearch
  - Significantly reduces response size and network overhead
  - Results still sorted by createdAt (ES handles sorting)
 
- **Performance Impact**:
  - Bulk Updates: Single ES refresh instead of N (where N = executions to cancel)
  - Query Efficiency: Only fetches IDs instead of full documents (~90%+ size reduction)
  - Network Efficiency: One bulk request instead of N individual update requests
  - Scalability: Handles large cancellation batches efficiently

### Code Deduplication & Simplification
- **Extracted Helper Method**: Created `checkConcurrencyIfNeeded()` helper method
  - Consolidates duplicated concurrency check logic from the different execution paths
  - Reduces code duplication and improves maintainability
  - Single source of truth for concurrency checking logic

- **Extracted TerminalExecutionStatuses Constant**: Improved code reuse and maintainability
  - Moved hardcoded terminal status arrays to a reusable `TerminalExecutionStatuses` constant in `v1.ts`
  - Replaced inline arrays across 5 files (17 occurrences) with the constant
  - Single source of truth for terminal statuses, reducing risk of inconsistencies
  - Better type safety with `readonly` array and proper TypeScript typing

### Initialization & Lifecycle
- **Proper Initialization Order**: Moved `ConcurrencyManager` initialization to `start()` method
  - Ensures all dependencies (`WorkflowTaskManager`, `WorkflowExecutionRepository`) are available
  - Follows Kibana plugin lifecycle patterns correctly

## Notes

- This PR implements only the `cancel-in-progress` strategy. Other strategies (`queue`, `drop`) will be implemented in subsequent PRs.
- The implementation cancels all non-terminal executions (PENDING, RUNNING, WAITING, etc.), not just RUNNING ones, as per the requirement that cancel-in-progress should cancel regardless of status.
- Unsupported Concurrency settings are temporarily hidden from autocomplete until all sub-tasks are complete (see residue task).

## Related Issues

Resolves https://github.com/elastic/security-team/issues/15056


## Screenshot

<img width="1860" height="697" alt="image" src="https://github.com/user-attachments/assets/321c06c4-7619-4c6c-a4ec-34e335d19f81" />

h88 committed:
implement cancel-in-progress collision strategy for workflow concurrency / init

h88 committed:
typecheck --fix

h88 committed:
Merge branch 'main' into 1wf-concurrency-params-subtask3-cancel-st

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/376421)
* Commit: 2f43d1f269ed0ba502e5082fc0cb9a3b60a5c376

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/376421#019b4adc-302c-4db4-a0c9-3d767e423ee3)

### History
* :broken_heart: [Build #375933](https://buildkite.com/elastic/kibana-pull-request/builds/375933) failed 399a512dd53029db0c25579b08d8cb2f2483bac6

cc @h88

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376421","number":376421,"commit":"2f43d1f269ed0ba502e5082fc0cb9a3b60a5c376"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375933","number":375933,"commit":"399a512dd53029db0c25579b08d8cb2f2483bac6"}],"number":376421}
buildkite-pr-comment-->

h88 committed:
Refactor: Extract reusable terminal execution statuses

h88 committed:
Merge branch 'main' into 1wf-concurrency-params-subtask3-cancel-st

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/376461)
* Commit: 2fc97edf3a7eab502c284a4e7034d8e043eea9f9

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/376461#019b4b3d-1389-472f-8860-9a448eda41f6)

### History
* :broken_heart: [Build #376421](https://buildkite.com/elastic/kibana-pull-request/builds/376421) failed 2f43d1f269ed0ba502e5082fc0cb9a3b60a5c376
* :broken_heart: [Build #375933](https://buildkite.com/elastic/kibana-pull-request/builds/375933) failed 399a512dd53029db0c25579b08d8cb2f2483bac6

cc @h88

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376461","number":376461,"commit":"2fc97edf3a7eab502c284a4e7034d8e043eea9f9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376421","number":376421,"commit":"2f43d1f269ed0ba502e5082fc0cb9a3b60a5c376"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375933","number":375933,"commit":"399a512dd53029db0c25579b08d8cb2f2483bac6"}],"number":376461}
buildkite-pr-comment-->

h88 committed:
terminal exec statuses refactor residue

h88 committed:
test --fix

h88 committed:
enable cancel-in-progress feat

h88 committed:
Refactor / inject dependencies and consolidate duplicate logic

h88 committed:
Merge branch 'main' into 1wf-concurrency-params-subtask3-cancel-st

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/376593)
* Commit: 8a1818c85e2b32989e3cc6f1da2ca94448853635

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/376593#019b4c67-ce76-4809-aa64-7444e9eb8ad7)

### History
* :green_heart: [Build #376522](https://buildkite.com/elastic/kibana-pull-request/builds/376522) succeeded dae96ef7c7b82e8efef9dd842fdd70561c678f82
* :broken_heart: [Build #376475](https://buildkite.com/elastic/kibana-pull-request/builds/376475) failed 3f04a5a2bea9f8bfd118207c8bdc4653ad438246
* :broken_heart: [Build #376461](https://buildkite.com/elastic/kibana-pull-request/builds/376461) failed 2fc97edf3a7eab502c284a4e7034d8e043eea9f9
* :broken_heart: [Build #376421](https://buildkite.com/elastic/kibana-pull-request/builds/376421) failed 2f43d1f269ed0ba502e5082fc0cb9a3b60a5c376
* :broken_heart: [Build #375933](https://buildkite.com/elastic/kibana-pull-request/builds/375933) failed 399a512dd53029db0c25579b08d8cb2f2483bac6

cc @h88

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376593","number":376593,"commit":"8a1818c85e2b32989e3cc6f1da2ca94448853635"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376522","number":376522,"commit":"dae96ef7c7b82e8efef9dd842fdd70561c678f82"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376475","number":376475,"commit":"3f04a5a2bea9f8bfd118207c8bdc4653ad438246"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376461","number":376461,"commit":"2fc97edf3a7eab502c284a4e7034d8e043eea9f9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376421","number":376421,"commit":"2f43d1f269ed0ba502e5082fc0cb9a3b60a5c376"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375933","number":375933,"commit":"399a512dd53029db0c25579b08d8cb2f2483bac6"}],"number":376593}
buildkite-pr-comment-->

h88 committed:
tests & types alignment with exposing only cancel-in-progress strategy atm

h88 committed:
Optimize concurrency cancellation with bulk updates and ID-only queries

- Add bulkUpdateWorkflowExecutions() method for efficient batch updates
- Optimize getRunningExecutionsByConcurrencyGroup() to return only IDs (string[])
- Add error handling for concurrency enforcement failures (best-effort approach)
- Update all tests and mocks to reflect optimizations

Performance improvements:
- Single ES refresh instead of N refreshes for bulk cancellations
- Reduced query response size (only IDs, not full documents)
- One network round trip for bulk updates instead of N

This reduces ES load and improves scalability when cancelling multiple
executions in cancel-in-progress collision strategy.

h88 committed:
Merge branch 'main' into 1wf-concurrency-params-subtask3-cancel-st

h88 committed:
*debug

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/376718)
* Commit: 2c611b4c7a76082700e52b7b78accc301d5c1691

### Failed CI Steps
* [FTR Configs #129](https://buildkite.com/elastic/kibana-pull-request/builds/376718#019b5106-0262-41ed-a56f-87bd7e2a49d4)
* [Scout: [ platform / streams_app ] plugin](https://buildkite.com/elastic/kibana-pull-request/builds/376718#019b510a-c303-456e-9937-b28b35173dbc)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/376718#019b5106-0262-41ed-a56f-87bd7e2a49d4) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/376718/jobs/019b5106-0262-41ed-a56f-87bd7e2a49d4/artifacts/019b512e-cc25-4a54-acb7-1b9af3bfb3c2) FTR Configs #<span></span>129 / Cases Attachment framework Attachment hooks Modal renders different solutions

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/db5f9543af9e23f284ef91571d180acddbe116b2) | [after](https://github.com/elastic/kibana/commit/2c611b4c7a76082700e52b7b78accc301d5c1691) | diff |
| --- | --- | --- | --- |
| `@kbn/workflows` | 371 | 372 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/db5f9543af9e23f284ef91571d180acddbe116b2) | [after](https://github.com/elastic/kibana/commit/2c611b4c7a76082700e52b7b78accc301d5c1691) | diff |
| --- | --- | --- | --- |
| `workflowsManagement` | 1.3MB | 1.3MB | -50.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/db5f9543af9e23f284ef91571d180acddbe116b2) | [after](https://github.com/elastic/kibana/commit/2c611b4c7a76082700e52b7b78accc301d5c1691) | diff |
| --- | --- | --- | --- |
| `@kbn/workflows` | 405 | 406 | +1 |


</details>


### History
* :broken_heart: [Build #376593](https://buildkite.com/elastic/kibana-pull-request/builds/376593) failed 8a1818c85e2b32989e3cc6f1da2ca94448853635
* :green_heart: [Build #376522](https://buildkite.com/elastic/kibana-pull-request/builds/376522) succeeded dae96ef7c7b82e8efef9dd842fdd70561c678f82
* :broken_heart: [Build #376475](https://buildkite.com/elastic/kibana-pull-request/builds/376475) failed 3f04a5a2bea9f8bfd118207c8bdc4653ad438246
* :broken_heart: [Build #376461](https://buildkite.com/elastic/kibana-pull-request/builds/376461) failed 2fc97edf3a7eab502c284a4e7034d8e043eea9f9
* :broken_heart: [Build #376421](https://buildkite.com/elastic/kibana-pull-request/builds/376421) failed 2f43d1f269ed0ba502e5082fc0cb9a3b60a5c376
* :broken_heart: [Build #375933](https://buildkite.com/elastic/kibana-pull-request/builds/375933) failed 399a512dd53029db0c25579b08d8cb2f2483bac6

cc @h88

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376718","number":376718,"commit":"2c611b4c7a76082700e52b7b78accc301d5c1691"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376593","number":376593,"commit":"8a1818c85e2b32989e3cc6f1da2ca94448853635"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376522","number":376522,"commit":"dae96ef7c7b82e8efef9dd842fdd70561c678f82"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376475","number":376475,"commit":"3f04a5a2bea9f8bfd118207c8bdc4653ad438246"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376461","number":376461,"commit":"2fc97edf3a7eab502c284a4e7034d8e043eea9f9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376421","number":376421,"commit":"2f43d1f269ed0ba502e5082fc0cb9a3b60a5c376"}],"number":376718}
buildkite-pr-comment-->

h88 committed:
Merge branch 'main' into 1wf-concurrency-params-subtask3-cancel-st

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/376854)
* Commit: 17cdee823312e6cfb83fb72b7ffdce28baa6795a

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/50eebddd6e1c339d5f6287b3a7238a2fa9568cd7) | [after](https://github.com/elastic/kibana/commit/17cdee823312e6cfb83fb72b7ffdce28baa6795a) | diff |
| --- | --- | --- | --- |
| `@kbn/workflows` | 371 | 372 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/50eebddd6e1c339d5f6287b3a7238a2fa9568cd7) | [after](https://github.com/elastic/kibana/commit/17cdee823312e6cfb83fb72b7ffdce28baa6795a) | diff |
| --- | --- | --- | --- |
| `workflowsManagement` | 1.3MB | 1.3MB | -50.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/50eebddd6e1c339d5f6287b3a7238a2fa9568cd7) | [after](https://github.com/elastic/kibana/commit/17cdee823312e6cfb83fb72b7ffdce28baa6795a) | diff |
| --- | --- | --- | --- |
| `@kbn/workflows` | 405 | 406 | +1 |


</details>


### History
* :yellow_heart: [Build #376718](https://buildkite.com/elastic/kibana-pull-request/builds/376718) was flaky 2c611b4c7a76082700e52b7b78accc301d5c1691
* :broken_heart: [Build #376593](https://buildkite.com/elastic/kibana-pull-request/builds/376593) failed 8a1818c85e2b32989e3cc6f1da2ca94448853635
* :green_heart: [Build #376522](https://buildkite.com/elastic/kibana-pull-request/builds/376522) succeeded dae96ef7c7b82e8efef9dd842fdd70561c678f82
* :broken_heart: [Build #376475](https://buildkite.com/elastic/kibana-pull-request/builds/376475) failed 3f04a5a2bea9f8bfd118207c8bdc4653ad438246
* :broken_heart: [Build #376461](https://buildkite.com/elastic/kibana-pull-request/builds/376461) failed 2fc97edf3a7eab502c284a4e7034d8e043eea9f9
* :broken_heart: [Build #376421](https://buildkite.com/elastic/kibana-pull-request/builds/376421) failed 2f43d1f269ed0ba502e5082fc0cb9a3b60a5c376

cc @h88

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376854","number":376854,"commit":"17cdee823312e6cfb83fb72b7ffdce28baa6795a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376718","number":376718,"commit":"2c611b4c7a76082700e52b7b78accc301d5c1691"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376593","number":376593,"commit":"8a1818c85e2b32989e3cc6f1da2ca94448853635"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376522","number":376522,"commit":"dae96ef7c7b82e8efef9dd842fdd70561c678f82"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376475","number":376475,"commit":"3f04a5a2bea9f8bfd118207c8bdc4653ad438246"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376461","number":376461,"commit":"2fc97edf3a7eab502c284a4e7034d8e043eea9f9"}],"number":376854}
buildkite-pr-comment-->

h88 committed:
Error scenario tests for ConcurrencyManager.checkConcurrency

h88 committed:
Merge branch 'main' into 1wf-concurrency-params-subtask3-cancel-st

h88 commented on src/platform/packages/shared/kbn-workflows/spec/lib/generate_yaml_schema_from_connectors.ts:
```diff
@@ -41,15 +41,13 @@ export function generateYamlSchemaFromConnectors(
-    settings: getWorkflowSettingsSchema(recursiveStepSchema, loose)
-      .omit({ concurrency: true })
-      .optional(),
+    settings: getWorkflowSettingsSchema(recursiveStepSchema, loose).optional(),
```
negative, it was removed from the origin type

h88 committed:
Fix type safety by normalizing EsWorkflowExecution before concurrency key evaluation + extract to local func and reuse

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/377133)
* Commit: bf4d1d325852293c7dbcad7ac13bda85d66ac11b

### Failed CI Steps
* [FTR Configs #1](https://buildkite.com/elastic/kibana-pull-request/builds/377133#019b6f64-11a9-472f-bd70-14bc065ea749)
* [FTR Configs #1](https://buildkite.com/elastic/kibana-pull-request/builds/377133#019b6f86-4259-4418-bb63-a6cfd9b1235b)
* [FTR Configs #80](https://buildkite.com/elastic/kibana-pull-request/builds/377133#019b6f64-11f6-4854-ad35-59a85a7984a8)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/377133#019b6f64-11f6-4854-ad35-59a85a7984a8) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/377133/jobs/019b6f64-11f6-4854-ad35-59a85a7984a8/artifacts/019b6f8d-0723-467d-ae5e-7597217c96d2) FTR Configs #<span></span>80 / Core Analysis - Entity Store @<span></span>ess Entity Store Engine APIs in non-default space get and list "before all" hook in "get and list"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/377133#019b6f64-11a9-472f-bd70-14bc065ea749) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/377133/jobs/019b6f64-11a9-472f-bd70-14bc065ea749/artifacts/019b6f86-259b-495d-ab5e-2304f8ba151d) FTR Configs #<span></span>1 / task_manager scheduling and running tasks should schedule a task with rrule with fixed time and dtstart
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/377133#019b6f86-4259-4418-bb63-a6cfd9b1235b) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/377133/jobs/019b6f86-4259-4418-bb63-a6cfd9b1235b/artifacts/019b6f91-4e8d-4379-9f6d-5a0d3c48f481) FTR Configs #<span></span>1 / task_manager scheduling and running tasks should schedule a task with rrule with fixed time and dtstart

### History
* :green_heart: [Build #376854](https://buildkite.com/elastic/kibana-pull-request/builds/376854) succeeded 17cdee823312e6cfb83fb72b7ffdce28baa6795a
* :yellow_heart: [Build #376718](https://buildkite.com/elastic/kibana-pull-request/builds/376718) was flaky 2c611b4c7a76082700e52b7b78accc301d5c1691
* :broken_heart: [Build #376593](https://buildkite.com/elastic/kibana-pull-request/builds/376593) failed 8a1818c85e2b32989e3cc6f1da2ca94448853635
* :green_heart: [Build #376522](https://buildkite.com/elastic/kibana-pull-request/builds/376522) succeeded dae96ef7c7b82e8efef9dd842fdd70561c678f82
* :broken_heart: [Build #376475](https://buildkite.com/elastic/kibana-pull-request/builds/376475) failed 3f04a5a2bea9f8bfd118207c8bdc4653ad438246
* :broken_heart: [Build #376461](https://buildkite.com/elastic/kibana-pull-request/builds/376461) failed 2fc97edf3a7eab502c284a4e7034d8e043eea9f9

cc @h88

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377133","number":377133,"commit":"bf4d1d325852293c7dbcad7ac13bda85d66ac11b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376854","number":376854,"commit":"17cdee823312e6cfb83fb72b7ffdce28baa6795a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376718","number":376718,"commit":"2c611b4c7a76082700e52b7b78accc301d5c1691"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376593","number":376593,"commit":"8a1818c85e2b32989e3cc6f1da2ca94448853635"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376522","number":376522,"commit":"dae96ef7c7b82e8efef9dd842fdd70561c678f82"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376475","number":376475,"commit":"3f04a5a2bea9f8bfd118207c8bdc4653ad438246"}],"number":377133}
buildkite-pr-comment-->

h88 committed:
Add validation to bulkUpdateWorkflowExecutions mock to match ES behaviour

h88 committed:
Merge branch 'main' into 1wf-concurrency-params-subtask3-cancel-st

h88 committed:
Explicit size param capped at ES default max_result_window; safe for ID-only query.

h88 committed:
Add guard checks for ConcurrencyManager initialization in task runners

h88 committed:
Merge branch 'main' into 1wf-concurrency-params-subtask3-cancel-st

talboren commented on src/platform/plugins/shared/workflows_execution_engine/server/concurrency/concurrency_manager.ts:
```diff
@@ -47,7 +54,6 @@ export class ConcurrencyManager {
 
     const keyExpression = concurrencySettings.key.trim();
     if (keyExpression === '') {
-      this.logger.debug('[Concurrency] Empty concurrency key provided');
```
why are we removing all those debug logs?

talboren commented on src/platform/packages/shared/kbn-workflows/spec/schema.test.ts:
```diff
@@ -328,9 +329,10 @@ describe('WorkflowSettingsSchema', () => {
     it('should accept all valid strategy values', () => {
-      expect(CollisionStrategySchema.safeParse('queue').success).toBe(true);
-      expect(CollisionStrategySchema.safeParse('drop').success).toBe(true);
+      // Only 'cancel-in-progress' is currently implemented; 'queue' and 'drop' are TBD
```
is this true? queue is the "default" we get ootb from task manager, right?

talboren reviewed (approved):
lgtm but left a few comments

h88 committed:
Merge branch 'main' into 1wf-concurrency-params-subtask3-cancel-st

h88 commented on src/platform/plugins/shared/workflows_execution_engine/server/plugin.ts:
```diff
@@ -578,4 +601,95 @@ export class WorkflowsExecutionEnginePlugin
+      // Best-effort concurrency enforcement: log error but allow execution to proceed
+      // This prevents a single cancellation failure from blocking new executions
+      const errorMessage = error instanceof Error ? error.message : String(error);
+      this.logger.debug(
```
debug because it's handled gracefully (best effort). Examples (like ES failure/etc) can be found in the unit tests (covered)

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/377679)
* Commit: fd8d9c313b975cb747c1777d338df5b47db37c6c

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/4a5bbde8c51c461317f6a03e4c2b94b82772cbbc) | [after](https://github.com/elastic/kibana/commit/fd8d9c313b975cb747c1777d338df5b47db37c6c) | diff |
| --- | --- | --- | --- |
| `@kbn/workflows` | 371 | 372 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/4a5bbde8c51c461317f6a03e4c2b94b82772cbbc) | [after](https://github.com/elastic/kibana/commit/fd8d9c313b975cb747c1777d338df5b47db37c6c) | diff |
| --- | --- | --- | --- |
| `workflowsManagement` | 1.3MB | 1.3MB | -50.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/4a5bbde8c51c461317f6a03e4c2b94b82772cbbc) | [after](https://github.com/elastic/kibana/commit/fd8d9c313b975cb747c1777d338df5b47db37c6c) | diff |
| --- | --- | --- | --- |
| `@kbn/workflows` | 405 | 406 | +1 |


</details>


### History
* :green_heart: [Build #377161](https://buildkite.com/elastic/kibana-pull-request/builds/377161) succeeded ea4fab312f31289a7994dd4909ae0cb8a9ab480c
* :broken_heart: [Build #377133](https://buildkite.com/elastic/kibana-pull-request/builds/377133) failed bf4d1d325852293c7dbcad7ac13bda85d66ac11b
* :green_heart: [Build #376854](https://buildkite.com/elastic/kibana-pull-request/builds/376854) succeeded 17cdee823312e6cfb83fb72b7ffdce28baa6795a
* :yellow_heart: [Build #376718](https://buildkite.com/elastic/kibana-pull-request/builds/376718) was flaky 2c611b4c7a76082700e52b7b78accc301d5c1691
* :broken_heart: [Build #376593](https://buildkite.com/elastic/kibana-pull-request/builds/376593) failed 8a1818c85e2b32989e3cc6f1da2ca94448853635
* :green_heart: [Build #376522](https://buildkite.com/elastic/kibana-pull-request/builds/376522) succeeded dae96ef7c7b82e8efef9dd842fdd70561c678f82

cc @h88

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377679","number":377679,"commit":"fd8d9c313b975cb747c1777d338df5b47db37c6c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377161","number":377161,"commit":"ea4fab312f31289a7994dd4909ae0cb8a9ab480c"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377133","number":377133,"commit":"bf4d1d325852293c7dbcad7ac13bda85d66ac11b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376854","number":376854,"commit":"17cdee823312e6cfb83fb72b7ffdce28baa6795a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376718","number":376718,"commit":"2c611b4c7a76082700e52b7b78accc301d5c1691"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376593","number":376593,"commit":"8a1818c85e2b32989e3cc6f1da2ca94448853635"}],"number":377679}
buildkite-pr-comment-->
